tools/sigclient/pkg/alerts/alerts_loadtest.go

```
@@ -0,0 +1,321 @@
+	startTime := time.Now()
+	summary := SummaryData{
+		MinuteData: make(map[int]MinuteSummaryData),
+	}
+
+	ticker := time.NewTicker(1 * time.Minute)
+	defer ticker.Stop()
+
+	timeoutDuration := 2 * time.Minute
+	timeout := time.NewTimer(timeoutDuration)
+	defer timeout.Stop()
+
+	for {
+		currentMinute := int(time.Since(startTime).Minutes())
+		select {
+		case webhookBody := <-webhookChan:
+			alertReceivedTime := time.Now()
+			timeout.Stop() // Stop the timeout as we received an alert
+			log.Infof("CurrentMinute=%v,ReceivedAlert=%v,NumEvaluations=%v", currentMinute, webhookBody.Title, webhookBody.NumEvaluationsCount)
+
+			alertEvalMinute := int(webhookBody.NumEvaluationsCount) - 1
+
+			minuteData, exists := summary.MinuteData[alertEvalMinute]
+			if !exists {
+				minuteData = MinuteSummaryData{
+					Alerts: make(map[string]AlertData),
+				}
+			}
+
+			delay := alertReceivedTime.Sub(startTime.Add(time.Duration(alertEvalMinute) * time.Minute))
+
+			alertData := minuteData.Alerts[webhookBody.Title]
+			alertData.ReceivedCount++
+			alertData.TotalDelay += delay
+			minuteData.Alerts[webhookBody.Title] = alertData
+			minuteData.TotalAlertsReceived++
+			minuteData.TotalDelay += delay
+
+			summary.TotalAlertsReceived++
+			summary.TotalDelay += delay
+
+			summary.MinuteData[alertEvalMinute] = minuteData
+
+			timeout.Reset(timeoutDuration) // Reset the timeout so that we won't trigger timeout and exit.
+		case <-ticker.C:
+			// Log summary of the previous minute
+			minute := currentMinute - 1
+			if minute >= 0 {
+				if minuteData, exists := summary.MinuteData[minute]; exists {
+					minuteData.AverageDelay = minuteData.TotalDelay / time.Duration(minuteData.TotalAlertsReceived)
+					logMinuteSummary(minute, minuteData, numAlerts)
+					summary.MinuteData[minute] = minuteData
+				}
+			}
+		case <-timeout.C:
+			log.Errorf("Timed out waiting for alerts")
+			doneChan <- false
+			return
```

nit: maybe rename `doneChan` to `successChan`, since I think you're actually sending the `success` state on it. The current version is a little confusing because here we timeout so we should be done but you send `false` on the channel

---

Yes, I want to throw an Error, if we timed out. I will change it to `successChan`.